easybuild/easyblocks/n/nvhpc.py

```diff
@@ -0,0 +1,201 @@
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.filetools import adjust_permissions, write_file
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root
+
+
+# contents for siterc file to make PGI/NVHPC pick up $LIBRARY_PATH
+# cfr. https://www.pgroup.com/support/link.htm#lib_path_ldflags
+SITERC_LIBRARY_PATH = """
+# get the value of the environment variable LIBRARY_PATH
+variable LIBRARY_PATH is environment(LIBRARY_PATH);
+
+# split this value at colons, separate by -L, prepend 1st one by -L
+variable library_path is
+default($if($LIBRARY_PATH,-L$replace($LIBRARY_PATH,":", -L)));
+
+# add the -L arguments to the link line
+append LDLIBARGS=$library_path;
+
+# also include the location where libm & co live on Debian-based systems
+# cfr. https://github.com/easybuilders/easybuild-easyblocks/pull/919
+append LDLIBARGS=-L/usr/lib/x86_64-linux-gnu;
+"""
+
+
+class EB_NVHPC(PackedBinary):
+    """
+    Support for installing the NVIDIA HPC SDK (NVHPC) compilers
+    """
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'default_cuda_version': ["10.1", "CUDA Version to be used as default (10.2, 11.0, ...)", CUSTOM],
+            'compute_capability':   ["70", "Compute Capability (70, 80, ...)", CUSTOM],
+            'module_byo_compilers': [False, "BYO Compilers: Remove compilers from module", CUSTOM],
+            'module_nvhpc_own_mpi': [False, "Add NVHPC's packaged OpenMPI to module", CUSTOM],
+            'module_add_math_libs': [False, "Add NVHPC's math libraries to module", CUSTOM],
+            'module_add_profilers': [False, "Add NVHPC's NVIDIA Profilers to module", CUSTOM],
+            'module_add_nccl':      [False, "Add NVHPC's NCCL library to module", CUSTOM],
+            'module_add_nvshmem':   [False, "Add NVHPC's NVSHMEM library to module", CUSTOM],
+            'module_add_cuda':      [False, "Add NVHPC's CUDA to module", CUSTOM]
+        }
+        return PackedBinary.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, define custom class variables specific to NVHPC."""
+        super(EB_NVHPC, self).__init__(*args, **kwargs)
+
+        self.nvhpc_install_subdir = os.path.join('Linux_x86_64', self.version)
+
+    def install_step(self):
+        """Install by running install command."""
+
+        nvhpc_env_vars = {
+            'NVHPC_INSTALL_DIR': self.installdir,
+            'NVHPC_SILENT': 'true',
+            'NVHPC_DEFAULT_CUDA': str(self.cfg['default_cuda_version']).lower(), #10.2, 11.0
+            'NVHPC_STDPAR_CUDACC': str(self.cfg['compute_capability']).lower(), #70, #80
+            }
+        cmd = "%s ./install" % ' '.join(['%s=%s' % x for x in sorted(nvhpc_env_vars.items())])
+        run_cmd(cmd, log_all=True, simple=True)
+
+        # make sure localrc uses GCC in PATH, not always the system GCC, and does not use a system g77 but gfortran
```

Comment:
I this comment still accurate? I need no mention of `gfortran` below?

Reply:
I just left it there from PGI, but I can have a look.

Reply:
It probably doesn't, so remove?

Reply:
Still needs to be cleaned up?

Reply:
What the regex does, it replaces 
```diff
29c29
< #PATH=/usr/bin:/bin:$PATH
---
> PATH=/usr/bin:/bin:$PATH
```
So I'm inclined to think this is still an issue as this places `/usr/bin` on top of everything else in `$PATH`. I tried to have a look if I could see any strange behavior when leaving the line in, but I don't know at which point in the chain NVHPC actually needs something from GCC and if takes the wrong one in that case…

Reply:
OK, fine, let's leave the comment in place.